src/jit/emitxarch.cpp

```diff
@@ -2903,14 +2903,20 @@ regNumber emitter::emitInsBinary(instruction ins, emitAttr attr, GenTree* dst, G
         varNum = tmpDsc->tdTempNum();
         offset = 0;
     }
+    else if (memBase->OperGet() == GT_LCL_VAR_ADDR)
+    {
+        varNum = memBase->AsLclVarCommon()->GetLclNum();
+        offset = 0;
```

Comment:
In this case, `memBase` is not actually a "base" in the usual sense. It is the full address (up above, we have set it to the op1 of the indir).
What isn't clear to me is how we can be sure that `memBase` is non-null at this point - is an indirection the only alternative to the above cases (in which case we would have set `memBase` as the op1 of the indir)?
In any event, I think it would be prudent to add a comment explaining why, and probably an assert.

---
In reply to: [88365005](https://github.com/dotnet/coreclr/pull/8097#discussion_r88365005) [](ancestors = 88365005)